heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -379,6 +393,102 @@ export class JestPlugin implements IHeftPlugin<IJestPluginOptions> {
+          propertyValue = path.resolve(configurationFileDir, restOfPath);
+        } else if ((packageDirMatches = PACKAGEDIR_REGEX.exec(propertyValue)) !== null) {
+          // Example:  <packageDir:@my/package>/path/to/file.js
+          const packageName: string | undefined = packageDirMatches.groups?.[PACKAGE_CAPTUREGROUP];
+          if (!packageName) {
+            throw new Error(
+              `Could not parse package name from "packageDir" token ` +
+                (propertyName ? `of property "${propertyName}" ` : '') +
+                `in "${configurationFilePath}".`
+            );
+          }
+
+          // Prevent module paths from being specified by limiting the number of '/' characters present
+          if (packageName.match(/^((@[^\/]+\/)|[^@])[^\/]+\/.*$/)) {
+            throw new Error(
+              `Module paths are not supported when using the "packageDir" token ` +
+                (propertyName ? `of property "${propertyName}" ` : '') +
+                `in "${configurationFilePath}". Only a package name is allowed.`
+            );
+          }
+
+          // Resolve to the root of the package (not the module referenced by the package). Substitute
+          // the heft-jest-plugin root if package name matches.
+          const resolvedPackagePath: string =
+            packageName === PLUGIN_PACKAGE_NAME
+              ? OWN_PACKAGE_FOLDER
+              : Import.resolvePackage({
+                  baseFolderPath: path.dirname(configurationFilePath),
+                  packageName
+                });
+          // First entry is the entire match
+          const restOfPath: string = path.normalize('./' + propertyValue.substr(packageDirMatches[0].length));
+          propertyValue = path.resolve(resolvedPackagePath, restOfPath);
+        }
+
+        // Return early, since the remainder of this function is used to resolve module paths
+        if (!options.resolveAsModule) {
+          return propertyValue;
+        }
+
+        // Example:  @rushstack/heft-jest-plugin
+        if (propertyValue === PLUGIN_PACKAGE_NAME) {
+          return OWN_PACKAGE_FOLDER;
+        }
+
+        // Example:  @rushstack/heft-jest-plugin/path/to/file.js
+        if (propertyValue.startsWith(PLUGIN_PACKAGE_NAME)) {
+          const restOfPath: string = path.normalize('./' + propertyValue.substr(PLUGIN_PACKAGE_NAME.length));
+          return path.join(OWN_PACKAGE_FOLDER, restOfPath);
+        }
+
+        return !options.modulePrefix
+          ? jestResolve(/*resolver:*/ undefined, {
+              rootDir: configurationFileDir,
+              filePath: propertyValue,
+              key: propertyName
+            })
+          : jestResolveWithPrefix(/*resolver:*/ undefined, {
+              rootDir: configurationFileDir,
+              filePath: propertyValue,
+              prefix: options.modulePrefix,
+              humanOptionName: propertyName,
```

Comment:
Jest's implementation specifies the `humanOptionName` as an English phrase -- do we care about that?
```js
  resolveWithPrefix(undefined, {
    filePath,
    humanOptionName: 'Test environment',
    optionName: 'testEnvironment',
    prefix: 'jest-environment-',
    requireResolveFunction,
    rootDir,
  });
```

Reply:
I'm not too concerned, no. Both of these fields are used for logging purposes, and setting them to the same value doesn't change the understanding of the logging message; if anything, it's more specific:
```
throw createValidationError(
    `  ${humanOptionName} ${chalk.bold(
      fileName,
    )} cannot be found. Make sure the ${chalk.bold(
      optionName,
    )} configuration option points to an existing node module.`,
  );
```